### ISSUE TITLE
[duckdb] Plural Intervals (days vs day, etc)

### ISSUE THREAD

**Issue by paultiq** [sqlfluff/sqlfluff#7374]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

DuckDB allows intervals to be defined with singular or plural forms of the Interval datatypes. 

SQLFluff rejects the plural forms. 

### Expected Behaviour

The following examples work with singular intervals (day/week/hour).

Simple example: 
```sql
select 
    interval 1 day as foo,
    interval 1 days as foo,
```
> L:   3 | P:  21 |  PRS | Line 3, Position 21: Found unparsable section: 'as foo,'

A more complete example would be: 
```sql
SELECT
  dts.start_time,
  count(*) OVER (ORDER BY dts.start_time) as num,
  count(*)
    OVER (
      ORDER BY
        dts.start_time
      RANGE BETWEEN INTERVAL 3 weeks PRECEDING AND interval 6 hours preceding
    ) AS num_roll
FROM
  range('2021-01-01'::date, '2021-01-02'::date, INTERVAL 1 HOUR)
    AS dts (start_time)
```
> L:   8 | P:   7 |  PRS | Line 8, Position 7: Found unparsable section: 'RANGE
>                       | BETWEEN INTERVAL 3 days PRECEDING ...'

### Observed Behaviour

SQLFluff fails to parse valid sql: 

> L: 8 | P: 7 | PRS | Line 8, Position 7: Found unparsable section: 'RANGE
> | BETWEEN INTERVAL 3 days PRECEDING ...'

### How to reproduce

`uv run sqlfluff lint --dialect duckdb test1.sql`

### Dialect

duckdb

### Version

sqlfluff, version 4.0.0

### Configuration

```

[tool.sqlfluff.core]
dialect = "duckdb"
templater = "jinja"
sql_file_exts = ".sql,.sql.j2,.dml,.ddl"
exclude_rules = [
    "AM04",
    "ST06",
    "LT01",
    "LT12",
    "JJ01"
]

[tool.sqlfluff.indentation]
tab_space_size = 2

[tool.sqlfluff.rules.capitalisation.keywords]
capitalisation_policy = "upper"
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
